include/glow/Runtime/StatsExporter.h

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_STATSEXPORTER_H
+#define GLOW_RUNTIME_STATSEXPORTER_H
+
+#include "llvm/ADT/StringRef.h"
+
+#include <vector>
+
+namespace glow {
+
+/// Interface for exporting runtime statistics.  The base implementation
+/// delegates to any subclass registered via `registerStatsExporter`.
+class StatsExporter {
+public:
+  /// Dtor.
+  virtual ~StatsExporter() {}
+
+  /// Add value to a time series.  May be called concurrently.
+  virtual void addTimeSeriesValue(llvm::StringRef key, double value) = 0;
+
+  /// Increment a counter.  May be called concurrently.
+  virtual void incrementCounter(llvm::StringRef key, int64_t value = 1) = 0;
+
+  /// Set a counter.  May be called concurrently.
+  virtual void setCounter(llvm::StringRef key, int64_t value) = 0;
+};
+
+/// Registry of StatsExporters.
+class StatsExporterRegistry final {
+public:
+  /// Dtor.
+  ~StatsExporterRegistry() {}
```

Comment:
why does this need to be declared?

Reply:
Oh.  It doesn't.  I'd originally derived this from StatsExporter, but after I did that I realized it was kind of useless.